src/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -2216,7 +2216,7 @@ public static void Select(IList checkRead, IList checkWrite, IList checkError, i
             // Validate input parameters.
             if ((checkRead == null || checkRead.Count == 0) && (checkWrite == null || checkWrite.Count == 0) && (checkError == null || checkError.Count == 0))
             {
-                throw new ArgumentNullException(SR.net_sockets_empty_select);
+                throw new ArgumentNullException(null, SR.net_sockets_empty_select);
```

Comment:
I actually tried this - but if you checkout the code, it only throws if **all** of the params are null and empty, not if only one is